js.go

```diff
@@ -1044,9 +1045,17 @@ func (js *js) subscribe(subj, queue string, cb MsgHandler, ch chan *Msg, isSync
 	if consumer != _EMPTY_ {
 		// Only create in case there is no consumer already.
 		info, err = js.ConsumerInfo(stream, consumer)
-		if err != nil && err.Error() != "nats: consumer not found" {
-			return nil, err
+		if err != nil {
+			if strings.Contains(err.Error(), "consumer not found") {
```

Comment:
This feels like could be simplified and possibly documented just a bit better as to what is going on here.

Reply:
Tried to refactor a bit this part